src/libraries/Microsoft.PowerFx.Core/Public/Types/TypeCheckers/AggregateTypeChecker.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Types.TypeCheckers
+{
+    internal abstract class AggregateTypeChecker
+    {
+        protected readonly ICollection<ExpressionError> _errorList;
+
+        public AggregateTypeChecker(ICollection<ExpressionError> errorList)
+        {
+            _errorList = errorList ?? throw new ArgumentNullException(nameof(errorList));
+        }
+
+        public bool Run(AggregateType sourceType, AggregateType typeToCheck, FormulaTypeChecker formulaTypeChecker)
+        {
+            if (sourceType == null || typeToCheck == null)
+            {
+                return false;
+            }
+            else if (sourceType._type.Kind != typeToCheck._type.Kind)
+            {
+                _errorList.Add(new ExpressionError()
+                {
+                    Kind = ErrorKind.Validation,
+                    ResourceKey = TexlStrings.ErrExpectedRVTypeMismatch,
+                    MessageArgs = new object[] { sourceType._type.GetKindString(), typeToCheck._type.GetKindString() }
+                });
+
+                return false;
+            }
+
+            return IsMatch(sourceType, typeToCheck, formulaTypeChecker);
+        }
+
+        public abstract bool IsMatch(AggregateType sourceType, AggregateType typeToCheck, FormulaTypeChecker formulaTypeChecker);
+
+        //public abstract bool IsFieldMatch(FormulaType sourceField, FormulaType fieldToCheck);
+    }
+
+    internal class StrictAggregateTypeChecker : AggregateTypeChecker
+    {
+        public StrictAggregateTypeChecker(ICollection<ExpressionError> errorList)
+            : base(errorList)
+        {
+        }
+
+        public override bool IsMatch(AggregateType sourceType, AggregateType typeToCheck, FormulaTypeChecker formulaTypeChecker)
+        {
+            var maybeDVEntitySource = sourceType.TableSymbolName;
+            var maybeDVEntityTarget = typeToCheck.TableSymbolName;
+
+            if ((maybeDVEntitySource != null || maybeDVEntityTarget != null) &&
+                maybeDVEntitySource != maybeDVEntityTarget)
+            {
+                _errorList.Add(new ExpressionError()
+                {
+                    Kind = ErrorKind.Validation,
+                    ResourceKey = TexlStrings.ErrExpectedRVRecordTableMismatch
```

Comment:
>ErrExpectedRVRecordTableMismatch [](http://example.com/codeflow?start=46&length=32)

can we include the names , this can be the same error we print for a scalar mismatch